### PR TITLE
Only select zones which are of type `availability-zone`

### PR DESCRIPTION
**Pull request by Skarlso**
### Description

Closes https://github.com/weaveworks/eksctl/issues/5110

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/eks/mocks/KubeNodeGroup.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Code generated by mockery v1.0.0. DO NOT EDIT.
```
Comment (Skarlso) [inline pkg/eks/mocks/KubeNodeGroup.go:1]: grrrr

**Review by cPu1 — COMMENTED** [review]
LGTM, but you'll need to rebase and migrate the SDK v1 EC2 types to v2.

**Comment by Skarlso** [comment]
Yeah. :D Just saw the conflicts.. :D

**Review by cPu1 — APPROVED** [review]

**Review by Himangini — APPROVED** [review]

**Review by LisaBrown3 — COMMENTED** [review]
#5002 
Duplicate of #
Duplicate of #

> ``

**Comment by cPu1** [comment]
> #5002 Duplicate of # Duplicate of #
> 
> > ``

@LisaBrown3 Did you mean to comment that on this issue?

**Pull request merged**